sdk/src/main/java/com/karhoo/sdk/api/service/loyalty/LoyaltyPointsToEarnInteractor.kt

```diff
@@ -0,0 +1,34 @@
+package com.karhoo.sdk.api.service.loyalty
+
+import com.karhoo.sdk.api.KarhooError
+import com.karhoo.sdk.api.datastore.credentials.CredentialsManager
+import com.karhoo.sdk.api.model.LoyaltyPoints
+import com.karhoo.sdk.api.network.client.APITemplate
+import com.karhoo.sdk.api.network.response.Resource
+import com.karhoo.sdk.api.service.common.BaseCallInteractor
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Inject
+import kotlin.coroutines.CoroutineContext
+
+internal class LoyaltyPointsToEarnInteractor @Inject constructor(credentialsManager: CredentialsManager,
```

Comment:
I have mixed feelings about having two separate interactors, we could have one which handles both earning and burning scenarios 
@mohadian what do you think?